src/smirnoffio.py

```diff
@@ -138,6 +137,44 @@ def build_pid(self, element, parameter):
             logger.info("Minor warning: Parameter ID %s doesn't contain any SMIRKS patterns, redundancies are possible\n" % ("/".join([InteractionType, parameter])))
             return "/".join([ParentType, InteractionType, parameter])
 
+def assign_openff_parameter(ff, new_value, pid):
+    """
+    Assign a SMIRNOFF parameter given the openforcefield.ForceField object, the desired parameter value,
+    and the parameter's unique ID.
+    """
+    # Split the parameter's unique ID into four fields using a slash:
+    # Input: ProperTorsions/Proper/k1/[*:1]~[#6X3:2]:[#6X3:3]~[*:4]
+    # Output: ProperTorsions, Proper, k1, [*:1]~[#6X3:2]:[#6X3:3]~[*:4]
+    # The first, third and fourth fields will be used for parameter assignment.
+    # We use "value_name" to describe names of individual numerical values within a single parameter type
+    # e.g. k1 in the above example.
+    (handler_name, tag_name, value_name, smirks) = pid.split('/')
+
+    # Get the OpenFF parameter object
+    parameter = ff.get_parameter_handler(handler_name).parameters[smirks]
+
+    if hasattr(parameter, value_name):
+        # If the value name is an attribute of the parameter then we set it directly.
+        unit = getattr(parameter, value_name).unit
+        setattr(parameter, value_name, new_value*unit)
+    else:
+        # If the value name is a periodic attribute (say, k1) then we need to use
+        # a regex to split the value name into 'k' and '1', then set the appropriate
+        # value in the k-list
+        attribute_split = re.split(r'(\d+)', value_name)
```

Comment:
I can't believe we didn't expose access to this in the original API. Thanks for being the crash test dummies on this interface, and sorry that it's coming up at such a critical phase!